src/app/core/auth/auth.service.ts

```diff
@@ -144,14 +145,29 @@ export class AuthService {
      * @param user
      */
     signUp(user: {
-        name: string;
-        email: string;
-        password: string;
-    }): Observable<any> {
-        return this._httpClient.post(
-            'http://localhost:3000/auth/register',
-            user
-        );
+      name: string;
+      surname: string;
+      email: string;
+      password: string;
+      role: IUserRole;
+    }): Promise<{ user: User; token: string } | undefined> {
+     const obs = this._httpClient.post<{ user: User; token: string}>(
+        'http://localhost:3000/auth/register',
+        user
+      );
+     obs.subscribe((res) => {
```

Comment:
you shouldn't be using Promises here.
To make your life easier and be more secure, you can just show a snackbar once the user creation is successful, but not move them forward. You need to let them login manually after they register, and the rest of the flow shall be untouched in this case.

Reply:
In general, that is better for us because it will be easier afterwards to add email confirmation etc